src/CoreWlan/CWKeychain.cs

```diff
@@ -0,0 +1,231 @@
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryFindWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, out NSString? username, out NSString? password)
+		{
+			username = null;
+			password = null;
+			NSStringRef usernamePtr = IntPtr.Zero;
+			NSStringRef passwordPtr = IntPtr.Zero;
+			var result = CWKeychainFindWiFiEAPUsernameAndPassword (domain, ssid.GetHandle (), out usernamePtr, out passwordPtr);
+			if (usernamePtr != IntPtr.Zero) {
+				username = Runtime.GetNSObject<NSString> (usernamePtr, true);
+			}
+			if (passwordPtr != IntPtr.Zero) {
+				password= Runtime.GetNSObject<NSString> (passwordPtr, true);
+			}
+			return result == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainFindWiFiPassword (CWKeychainDomain domain, NSDataRef ssid, out NSStringRef password);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TryFindWiFiPassword (CWKeychainDomain domain, NSData ssid, out NSString? password)
+		{
+			password = null;
+			NSStringRef passwordPtr = IntPtr.Zero;
+			var result = CWKeychainFindWiFiPassword (domain, ssid.GetHandle (), out passwordPtr);
+			if (passwordPtr != IntPtr.Zero) {
+				password= Runtime.GetNSObject<NSString> (passwordPtr, true);
+			}
+			return result == 0;
+		}
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainSetWiFiEAPIdentity (CWKeychainDomain domain, NSDataRef ssid, SecIdentityRef identity);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TrySetWiFiEAPIdentity (CWKeychainDomain domain, NSData ssid, SecIdentity? identity)
+			=> CWKeychainSetWiFiEAPIdentity (domain, ssid.GetHandle (), identity.GetHandle ()!) == 0;
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		[DllImport (Constants.CoreWlanLibrary)]
+		static extern OSStatus CWKeychainSetWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSDataRef ssid, NSStringRef username, NSStringRef password);
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10,9)]
+#endif
+		public static bool TrySetWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, NSString? username, NSString? password)
+			=> CWKeychainSetWiFiEAPUsernameAndPassword (domain, ssid.GetHandle (), username.GetHandle ()!, password.GetHandle ()!) == 0;
+
+#if NET
+		[SupportedOSPlatform ("macos10.9")]
+#else
+		[Mac (10, 9)]
+#endif
+		public static bool TrySetWiFiEAPUsernameAndPassword (CWKeychainDomain domain, NSData ssid, string? username, string? password)
+		{
+			using (NSString? nsUsername = (username == null)? null : new NSString (username))
+			using (NSString? nsPassword = (password == null)? null : new NSString (password)) {
```

Comment:
You can use `CFString.CreateNative` instead of creating a managed NSString wrapper. It's much more efficient.